### PR TITLE
Update lint dependencies

### PR DESCRIPTION
**Pull request by bmish**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -126,15 +126,15 @@
     "ember-cli-dependency-checker": "^3.3.1",
     "ember-cli-yuidoc": "^0.9.1",
     "ember-publisher": "0.0.7",
-    "eslint": "^8.28.0",
+    "eslint": "^8.29.0",
     "eslint-config-prettier": "^8.5.0",
     "eslint-import-resolver-node": "^0.3.6",
     "eslint-plugin-disable-features": "^0.1.3",
-    "eslint-plugin-ember-internal": "^2.0.0",
+    "eslint-plugin-ember-internal": "^3.0.0",
```
Comment (bmish) [inline package.json:133]: https://github.com/emberjs/eslint-plugin-ember-internal/releases/tag/v3.0.0

File: .eslintrc.js [hunk 2]
```diff
@@ -153,13 +153,14 @@ module.exports = {
       },
 
       plugins: ['node'],
-      rules: Object.assign({}, require('eslint-plugin-node').configs.recommended.rules, {
+      extends: ['plugin:node/recommended'],
+      rules: {
```
Comment (bmish) [inline .eslintrc.js:157]: Modern way of writing this. Same behavior.

**Review by chriskrycho — COMMENTED** [review]
Thank you!

**Comment by chriskrycho** [comment]
@bmish unfortunately, the smoke tests are failing on linting: https://github.com/emberjs/ember.js/actions/runs/3691613903/jobs/6249825386#step:6:62

**Comment by bmish** [comment]
Removed change to smoke-tests since those need more work to update to ESLint v8.

**Comment by chriskrycho** [comment]
Makes sense. Thank you!

**Pull request merged**